src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,258 @@
+      m_elevatorSim =
+          new ElevatorSim(
+              kElevatorGearbox,
+              kElevatorGearing,
+              kCarriageMass.in(Kilograms),
+              kElevatorDrumRadius.in(Meters),
+              kMinElevatorHeight.in(Meters),
+              kMaxElevatorHeight.in(Meters),
+              true,
+              0);
+      elevatorMech =
+          mech.getRoot("root", 1, 0)
+              .append(
+                  new MechanismLigament2d(
+                      "Elevator Mech [0]", 1, 90, 10, new Color8Bit(Color.kPurple)));
+
+      if (widg == null) {
+        widg = Shuffleboard.getTab(getName()).add("Elevator", mech);
+      }
+    }
+
+    SmartDashboard.putNumber("Elevator/Setpoint", 0);
+    SmartDashboard.putNumber("Elevator/Setpoint Rotations", 0);
+  }
+
+  public static Distance rotationsToHeight(Angle rotations) {
+    return ElevatorConstants.kElevatorDrumCircumference
+        .times(rotations.in(Rotations) * ElevatorConstants.kCarageFactor)
+        .div(kElevatorGearing);
+  }
+
+  // height = C * rot * 2/75 -> rot = height * 75/2C
+  public static Angle heightToRotations(Distance height) {
+    return height
+        .times(kElevatorGearing)
+        .div(kElevatorDrumCircumference.times(kCarageFactor))
+        .times(Rotations.one());
+  }
+
+  public static AngularVelocity heightToRotations(LinearVelocity vel) {
+    // rot/s = (G)/(C*2/vel)
+    return Rotations.one()
+        .times(kElevatorGearing)
+        .div((kElevatorDrumCircumference.times(kCarageFactor)).div(vel));
+  }
+
+  public Distance getElevatorHeight() {
+    return rotationsToHeight(elevatorMotor1.getPosition().getValue());
+  }
+
+  public Command goUp() {
+    return startEnd(
+        () -> {
+          elevatorMotor1.set(elevatorOutput);
+        },
+        () -> elevatorMotor1.set(0));
+  }
+
+  private void disableSoftLimits() {
+    elevatorMotor1
+        .getConfigurator()
+        .apply(
+            new SoftwareLimitSwitchConfigs()
+                .withForwardSoftLimitEnable(false)
+                .withReverseSoftLimitEnable(false));
+  }
+
+  private void enableSoftLimits() {
+    elevatorMotor1.getConfigurator().apply(elvSoftLimit);
+  }
+
+  public Command limitHit() {
+    return runOnce(this::disableSoftLimits)
+        .andThen(goDown().until(elvLimitSwitch::get))
+        .andThen(zeroMotorEncoder())
+        .andThen(runOnce(this::enableSoftLimits));
+  }
+
+  public Command goDown() {
+    return startEnd(
+        () -> {
+          elevatorMotor1.set(-elevatorOutput);
+        },
+        () -> elevatorMotor1.set(0));
+  }
+
+  public Command zeroMotorEncoder() {
+    return runOnce(
+        () -> {
+          elevatorMotor1.setPosition(0);
+        });
+  }
+
+  public Command changeElevation(Distance heightLevel) {
+    return runOnce(
+        () -> {
+          Angle adjustedSetpoint = heightToRotations(heightLevel);
+          elevatorMotor1.setControl(new MotionMagicVoltage(adjustedSetpoint));
```

Comment:
Can we at a if-else to set it to PositionVoltage for simulation only so it moves in sim?